UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -128,10 +132,32 @@ public async Task<List<ShowDailyFilesResponseModel>> GetDailyBatchDetailsFiles(s
                 _logger.LogError(EventIds.ShowDailyFilesResponseFailed.ToEventId(), "Maritime safety information request to get daily NM files failed to return data with exception:{exceptionMessage} for _X-Correlation-ID:{CorrelationId}", ex.Message, correlationId);
                 throw;
             }
-
             return showDailyFilesResponses;
-
         }
 
+        public async Task<ShowWeeklyFilesResponseModel> GetWeeklyFilesResponseModelsAsync(int year, int week, string correlationId)
+        {
+            _logger.LogInformation(EventIds.GetWeeklyFilesResponseStarted.ToEventId(), "Maritime safety information request to get weekly NM files response started with _X-Correlation-ID:{correlationId}", correlationId);
+            ShowWeeklyFilesResponseModel showWeeklyFilesResponses = new();
+            try
+            {
+                showWeeklyFilesResponses.YearAndWeek = await GetAllYearWeek(correlationId);
+                if (year == 0 && week == 0)
+                {
+                    _logger.LogInformation(EventIds.GetWeeklyFilesResponseForYearAndWeekWithZero.ToEventId(), "Maritime safety information request to get weekly NM files response for year and week with zero with _X-Correlation-ID:{correlationId}", correlationId);
+
+                    year = Convert.ToInt32(showWeeklyFilesResponses.YearAndWeek.OrderByDescending(x => x.Year).Select(x => x.Year).FirstOrDefault());
+                    week = Convert.ToInt32(showWeeklyFilesResponses.YearAndWeek.OrderByDescending(x => x.Week).Select(x => x.Week).FirstOrDefault());
+                }
+                showWeeklyFilesResponses.ShowFilesResponseModel = await GetWeeklyBatchFiles(year, week, correlationId);
```

Comment:
Please add log for request completed.
_logger.LogInformation(EventIds.GetWeeklyFilesResponseCompleted.ToEventId(), "Maritime safety information request to get weekly NM files response completed with _X-Correlation-ID:{correlationId}", correlationId);

Reply:
Added GetWeeklyFilesResponseCompleted log